components/NavBar.vue

```diff
@@ -23,11 +23,20 @@
 export default {
   data() {
     return {
+      searchText: "",
+      searchTimeout: "",
       logo: require("@/assets/img/leaf.svg"),
       searchIcon: require("@/assets/img/search.svg"),
-    };
+    }
   },
   methods: {
+    search() {
+      clearTimeout(this.searchTimeout)
+      this.searchTimeout = setTimeout(this.performSearch, 800)
```

Comment:
Please use debounce from lodash

Reply:
Vai Ishrat apu instructed us not to use lodash

Reply:
Then we can create a debounce function using raw JS and use it here.

Reply:
resolved